libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,14 +13,273 @@
+                Parsed transcription from the audio file.
+
+        Raises:
+            ImportError:
+                If the required packages `openai` or `pydub` are not installed.
+            ValueError:
+                If the file type is unsupported or exceeds the size limit.
+        """
+
+        try:
+            import openai
+        except ImportError:
+            raise ImportError(
+                "openai package not found, please install it with "
+                "`pip install openai`"
+            )
+
+        try:
+            from pydub import AudioSegment
+        except ImportError:
+            raise ImportError(
+                "pydub package not found, please install it with " "`pip install pydub`"
+            )
+
+        if not str(blob.path).endswith(self.input_format):
+            raise ValueError(
+                "File must be of one of the following types: "
+                f"{str(self.input_format)}"
+            )
+
+        if os.path.getsize(str(blob.source)) > self.byte_limit:
+            raise ValueError(
+                "Audio file exceeds 25 MB limit,"
+                f" length of file is {os.path.getsize(str(blob.source))} bytes"
+            )
+
+        if is_openai_v1():
+            client = openai.AzureOpenAI(
+                api_key=self.api_key,
+                azure_endpoint=self.azure_endpoint,
+                api_version=self.api_version,
+                max_retries=self.max_retries,
+                azure_ad_token=self.azure_ad_token,
+            )
+        else:
+            # Set the API key if provided
+            if self.api_key:
+                openai.api_key = self.api_key
+            if self.azure_endpoint:
+                openai.base_url = self.azure_endpoint
+
+        # Audio file from disk
+        audio = AudioSegment.from_file(blob.path)
+        # Get file extension from the file path
+        dot_file_extension = os.path.splitext(str(blob.path))[1]
+        file_extension = dot_file_extension[1:]
+        # m4a and mpga are not valid format flags for ffmpeg
+        # They do have equivalent flags that can be used though
+        format_map = {"m4a": "mp4", "mpga": "mp3"}
+        file_extension = format_map.get(file_extension, file_extension)
+
+        file_obj = io.BytesIO(audio.export(format=file_extension).read())
+
+        if blob.source is not None:
+            file_obj.name = str(blob.source)
+        else:
+            file_obj.name = str(blob.path)
+
+        # Transcribe
+        try:
+            if is_openai_v1():
+                transcript = client.audio.transcriptions.create(
+                    model=self.deployment_name,
+                    file=file_obj,
+                    **self._create_params,
+                )
+            else:
+                transcript = openai.Audio.transcribe(self.deployment_name, file_obj)
+        except Exception:
+            raise
+
+        yield Document(
+            page_content=transcript.text,
```

Comment:
I think we need this same check here as in the openai whisper parser:

page_content=transcript.text
                if not isinstance(transcript, str)
                else transcript,


It seems weird, but if a user selects response_format="text", the transcript returned will be `str` instead of an object with a text property

Reply:
Did not know that! has been updated to:

```python
yield Document(
                page_content=transcript.text
                if not isinstance(transcript, str)
                else transcript,
                metadata={"source": blob.source},
            )
```